xpk.py

```diff
@@ -4072,6 +4076,183 @@ def install_nccl_on_cluster(args) -> int:
   return 0
 
 
+@dataclass
+class GkeServerConfig:
+  """Stores the valid gke versions based on gcloud recommendations."""
+
+  default_rapid_gke_version: str
+  valid_master_versions: set[str]
+  valid_node_versions: set[str]
+
+
+def get_gke_server_config(args) -> tuple[int, GkeServerConfig | None]:
+  """Determine the GKE versions supported by gcloud currently.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    Tuple of
+    int: 0 if successful and 1 otherwise.
+    GkeServerConfig: stores valid gke version to use in node pool and cluster.
+  """
+  base_command = (
+      'gcloud container get-server-config'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+  )
+  default_rapid_gke_version_cmd = (
+      base_command
+      + " | awk '/RAPID/,/defaultVersion/' | awk 'FNR == 2 {print $2}' "
+  )
+  valid_master_versions_cmd = (
+      base_command
+      + " | awk '/validMasterVersions/,/validNodeVersions/' | awk '/-/,/-/"
+      " {print $2}' "
+  )
+  valid_node_versions_cmd = (
+      base_command
+      + " | awk '/validNodeVersions/,/END/' | awk '/-/,/-/ {print $2}' "
+  )
+  base_command_description = 'Determine server supported GKE versions for'
+
+  task = 'default rapid gke version'
+  return_code, default_rapid_gke_version = run_command_for_value(
```

Comment:
Can we move these 3 run_command_for_value() into a single method that reduces human error in the commands, logs etc.
Something like this:
```
gke_version_command_map = {}
gke_version_command_map['default rapid gke version'] =  default_rapid_gke_version_cmd
for task_name, command in gke_version_command_map.items():
      return_code, gke_version = run_command_for_value(
           command, f'{base_command_description} {task_name}.', args)
      if return_code != 0:
         xpk_print(f'Unable to get server config for {task}.')
      return return_code, None
```

Reply:
Done! Thanks for the suggestion!